vrpn_3DConnexion.C

```diff
@@ -230,98 +232,189 @@ void vrpn_3DConnexion::decodePacket(size_t bytes, vrpn_uint8 *buffer)
   // regardless of how many bytes were in the report.  This is going to get us into trouble for
   // multi-report packets.  Instead, we should go until we've parsed all characters and add the
   // number of characters parsed each time rather than a constant 7 reports.
+  if(bytes<13) bytes=13;
```

Comment:
This makes the next line meaningless.  Also, I think it may break the driver for 7-byte devices.  I think that the approach that was in the code of saying there are 7 bytes when there are not seems wrong (as indicate by the XXX).  I'd say it makes sense to change the code so that it continues so long as there is a byte left, first checking the  report type and then parsing and skipping the correct number of bytes.  It looks like there will need to be a different parsing function for the 13-byte report because it seems to be using the same report type for that as the other devices use for a 7-byte report.  So the type-1 reports are different on the different devices.

The derived classes that take 13-byte reports would override the decodePacket() method and change the 7s to 13s.

Reply:
Yes I noticed it breaks the 7-byte driver so i closed it. I did not notice, because I only have the Pro at home and the Compact at work so I can't test both of them simultaneously.

I will do it from scratch and make a pull request on the main repository when it is finished.